packages/job-manager/src/batch/batch.spec.ts

```diff
@@ -65,124 +83,59 @@ describe(Batch, () => {
+            items1.odatanextLink = 'odatanextLink-1';
+            const items2 = new JobListStub([{ id: 'job-id-2' }]);
+            jobMock
+                .setup(async o => o.list(options))
+                .returns(async () => Promise.resolve(<JobListResponse>(<unknown>items1)))
                 .verifiable();
-
-            await batch.waitJob(jobId1, 200);
-
-            taskMock.verify(async t => t.list(jobId1, It.isAny()), Times.atLeast(3));
-        });
-    });
-
-    describe('getCreatedTasksState()', () => {
-        it('get created job tasks state with pagination', async () => {
-            const cloudTaskListResultFirst = <BatchServiceModels.CloudTaskListResult>[];
-            cloudTaskListResultFirst.odatanextLink = 'nextPageLink';
-            cloudTaskListResultFirst.push({
-                id: 'job-1',
-                state: JobTaskState.completed,
-                executionInfo: <BatchServiceModels.TaskExecutionInformation>{
-                    result: JobTaskExecutionResult.success,
-                },
-            });
-            const cloudTaskListResultNext: BatchServiceModels.CloudTaskListResult = <BatchServiceModels.CloudTaskListResult>[];
-            cloudTaskListResultNext.push({
-                id: 'job-2',
-                state: JobTaskState.completed,
-                executionInfo: <BatchServiceModels.TaskExecutionInformation>{
-                    result: JobTaskExecutionResult.success,
-                },
-            });
-            const jobTasksExpected = [
-                {
-                    id: 'job-1',
-                    state: JobTaskState.completed,
-                    result: JobTaskExecutionResult.success,
-                    correlationId: 'messageId-1',
-                },
-                {
-                    id: 'job-2',
-                    state: JobTaskState.completed,
-                    result: JobTaskExecutionResult.success,
-                    correlationId: 'messageId-2',
-                },
-            ];
-            let i = 0;
-            const messages = [
-                {
-                    messageId: 'messageId-1',
-                    messageText: '{}',
-                },
-                {
-                    messageId: 'messageId-2',
-                    messageText: '{}',
-                },
-            ];
-
-            taskMock
-                .setup(async o => o.list(jobId1))
-                .returns(async () => {
-                    if (i++ === 0) {
-                        return Promise.resolve(cloudTaskListResultFirst as any);
-                    } else {
-                        return Promise.resolve(<BatchServiceModels.CloudTaskListResult>{});
-                    }
-                })
+            jobMock
+                .setup(async o => o.listNext(items1.odatanextLink, options))
+                .returns(async () => Promise.resolve(<JobListResponse>(<unknown>items2)))
                 .verifiable();
```

Comment:
create a helper function that sets up jobs given a list of job ids (& P2: may be number of pages)

Reply:
won't fix, unnecessary complication of code & make it harder to read test